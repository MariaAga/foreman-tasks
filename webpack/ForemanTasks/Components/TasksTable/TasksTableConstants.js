import { getControllerSearchProps } from 'foremanReact/constants';

export const TASKS_TABLE_ID = 'TASKS_TABLE';
export const TASKS_CONTROLLER = 'tasks';
export const TASKS_SUCCESS = 'TASKS_SUCCESS';
export const TASKS_REQUEST = 'TASKS_REQUEST';
export const TASKS_FAILURE = 'TASKS_FAILURE';
export const SELECT_ROWS = 'SELECT_ROWS';
export const UNSELECT_ROWS = 'UNSELECT_ROWS';
export const UNSELECT_ALL_ROWS = 'UNSELECT_ALL_ROWS';
export const CANCEL = 'CANCEL';
export const RESUME = 'RESUME';
export const CLOSED = 'CLOSED';

export const TASKS_TABLE_SELECTED_MODAL = 'TASKS_TABLE_SELECTED_MODAL';
export const TASKS_SEARCH_PROPS = getControllerSearchProps('tasks');
