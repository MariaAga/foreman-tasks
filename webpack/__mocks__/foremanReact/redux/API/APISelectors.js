export const selectAPIResponse = (state, key) => ({
  data: {
    text: 'some-data',
    key,
  },
});

export const selectAPIStatus = (state, key) => 'PENDING';
export const selectAPIError = (state, key) => ({ error: `${key} ERRROR` });
