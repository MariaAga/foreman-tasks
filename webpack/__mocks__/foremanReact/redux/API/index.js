export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};

export const get = data => ({ type: 'some-type', ...data });
