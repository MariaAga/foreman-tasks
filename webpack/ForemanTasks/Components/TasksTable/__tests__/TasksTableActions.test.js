import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/API';
import { TASKS_TABLE_ID } from '../TasksTableConstants';
import {
  getTableItems,
  cancelTask,
  cancelTaskRequest,
  resumeTask,
  resumeTaskRequest,
  bulkCancel,
  bulkResume,
} from '../TasksTableActions';

jest.mock('foremanReact/components/common/table', () => ({
  getTableItemsAction: jest.fn(controller => controller),
}));

jest.mock('foremanReact/API');

API.post.mockImplementation(() => ({ data: 'some-data' }));

const taskInfo = {
  id: 'some-id',
  name: 'some-name',
};

const fixtures = {
  'should cancelTask': () => cancelTask({ ...taskInfo, url: 'some-url' }),
  'should cancelTaskRequest and succeed': () =>
    cancelTaskRequest('some-id', 'some-name'),
  'should cancelTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return cancelTaskRequest('some-id', 'some-name');
  },

  'should resumeTask': () => resumeTask({ ...taskInfo, url: 'some-url' }),
  'should resumeTaskRequest and succeed': () => {
    API.post.mockImplementation(() => ({ data: 'some-data' }));
    return resumeTaskRequest('some-id', 'some-name');
  },
  'should resumeTaskRequest and fail': () => {
    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return resumeTaskRequest('some-id', 'some-name');
  },
  'handles bulkResume requests that fail': () => {
    const selected = [{ ...taskInfo, isResumable: true, isCancellable: false }];

    API.post.mockImplementation(() =>
      Promise.reject(new Error('Network Error'))
    );
    return bulkResume({ selected, url: 'some-url' });
  },
  'handles resumable bulkResume requests': () => {
    const selected = [{ ...taskInfo, isResumable: true, isCancellable: false }];

    API.post.mockImplementation(() => ({
      data: {
        resumed: [{ action: 'I am resumed' }],
        failed: [{ action: 'I am failed' }],
      },
    }));
    return bulkResume({ selected, url: 'some-url' });
  },
  'handles bulkCancel requests': () => {
    const selected = [{ ...taskInfo, isResumable: false, isCancellable: true }];
    return bulkCancel({ selected, url: 'some-url' });
  },
  'handles bulkCancel requests that are not cancellable': () => {
    const selected = [
      { ...taskInfo, isResumable: false, isCancellable: false },
    ];
    return bulkCancel({ selected, url: 'some-url' });
  },
  'handles bulkResume requests that are not resumable': () => {
    const selected = [
      { ...taskInfo, isResumable: false, isCancellable: false },
    ];
    return bulkResume({ selected, url: 'some-url' });
  },
};
describe('TasksTable actions', () => {
  it('getTableItems should reuse common/table/getTableItemsAction', () => {
    expect(getTableItems('')).toEqual(TASKS_TABLE_ID);
  });
  testActionSnapshotWithFixtures(fixtures);
});
