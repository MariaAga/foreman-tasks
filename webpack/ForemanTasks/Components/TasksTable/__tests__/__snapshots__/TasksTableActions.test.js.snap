// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTable actions handles bulkCancel requests 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    "TASKS_TABLE",
  ],
  Array [
    [Function],
  ],
]
`;

exports[`TasksTable actions handles bulkCancel requests that are not cancellable 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Not all the selected tasks can be canceled",
          "type": "warning",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions handles bulkResume requests that are not resumable 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Not all the selected tasks can be resumed",
          "type": "warning",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions handles bulkResume requests that fail 1`] = `
Array [
  Array [
    Object {
      "type": "TASKS_RESUME_REQUEST",
    },
  ],
  Array [
    Object {
      "error": [Error: Network Error],
      "type": "TASKS_RESUME_FAILURE",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Cannot resume tasks at the moment Error: Network Error",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions handles resumable bulkResume requests 1`] = `
Array [
  Array [
    Object {
      "type": "TASKS_RESUME_REQUEST",
    },
  ],
  Array [
    Object {
      "type": "TASKS_RESUME_SUCCESS",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "I am resumed Task execution was resumed",
          "type": "success",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "I am failed Task execution could not be resumed",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    "TASKS_TABLE",
  ],
  Array [
    [Function],
  ],
]
`;

exports[`TasksTable actions should cancelTask 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    "TASKS_TABLE",
  ],
  Array [
    [Function],
  ],
]
`;

exports[`TasksTable actions should cancelTaskRequest and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Trying to cancel \\"some-name\\" task",
          "type": "info",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task cannot be cancelled at the moment.",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions should cancelTaskRequest and succeed 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Trying to cancel \\"some-name\\" task",
          "type": "info",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task cancelled",
          "type": "success",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions should resumeTask 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    "TASKS_TABLE",
  ],
  Array [
    [Function],
    undefined,
  ],
]
`;

exports[`TasksTable actions should resumeTaskRequest and fail 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Task \\"some-name\\" has to be resumable.",
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;

exports[`TasksTable actions should resumeTaskRequest and succeed 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "\\"some-name\\" Task execution was resumed",
          "type": "success",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;
